samples/word/04-range/split-words-of-first-paragraph.yaml

```diff
@@ -0,0 +1,161 @@
+id: word-basics-split-words-of-first-paragraph-
+name: Range manipulation
+description: 'Show how to split a paragraph into word ranges and then traverses all the ranges to format each word, producing a "karaoke" effect.'
+author: Microsoft
+host: WORD
+api_set:
+    WordApi: 1.1
+script:
+    content: |-
+        $("#setup").click(setup);
+        $("#highlight").click(highlightWords);
+
+        function highlightWords() {
+            Word.run(function (context) {
+                // var myWords = context.document.body.paragraphs.first.split([" "], true, true);
+                var myPars = context.document.body.paragraphs;
+                context.load(myPars, { expand: 'font' });
+                return context.sync()
+                    .then(function () {
+                        var myWords = myPars.items[0].split([" "], true, true);
+                        context.load(myWords, { expand: 'font' });
+                        return context.sync()
+                            .then(function () {
+                                return forEach(myWords, function (item, i) {
+                                    if (i >= 1) {
+                                        myWords.items[i - 1].font.highlightColor = "#FFFFFF";
+                                    }
+                                    myWords.items[i].font.highlightColor = "#FFFF00";
+                                    return createTimerPromise(200).then(context.sync);
+                                })
+                            })
+                    })
+            })
+                .catch(OfficeHelpers.Utilities.log);
+
+            function createTimerPromise(ms) {
+                return new OfficeExtension.Promise(function (resolve) {
+                    setTimeout(resolve, ms);
+                })
+            }
+
+            function forEach(collection, handler) {
+                var promise = new OfficeExtension.Promise(function (resolve) { resolve(); });
```

Comment:
BTW, this code will fail on RTM, where we don't (yet) expose OfficeExtension.Promise.  you can use a regular "Promise" instead, though.  Or use async/await, which uses "Promise" under the covers.